cors.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2021-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package connect
+
+import (
+	"net/http"
+
+	"github.com/rs/cors"
```

Comment:
Not sure how we feel about bringing in dependencies, but given the usage here is pretty straightforward, we can/should easily be able to implement the subset of features we need to expose imo without bringing in a dependency. CORS is relatively simple to implement.